ai/ollama/v0/client.go

```diff
@@ -0,0 +1,174 @@
+	Details    struct {
+		Format            string `json:"format"`
+		Family            string `json:"family"`
+		Families          string `json:"families"`
+		ParameterSize     string `json:"parameter_size"`
+		QuantizationLevel string `json:"quantization_level"`
+	} `json:"details"`
+}
+
+type ListLocalModelsRequest struct {
+}
+
+type ListLocalModelsResponse struct {
+	Models []OllamaModelInfo `json:"models"`
+}
+
+func (c *OllamaClient) CheckModelAvailability(modelName string) bool {
+	request := &ListLocalModelsRequest{}
+	response := &ListLocalModelsResponse{}
+	req := c.httpClient.R().SetResult(&response).SetBody(request)
+	if _, err := req.Get("/api/tags"); err != nil {
+		return false
+	}
+	localModels := []string{}
+	for _, m := range response.Models {
+		localModels = append(localModels, m.Name)
+	}
+	return slices.Contains(localModels, modelName)
+}
+
+type PullModelRequest struct {
+	Name   string `json:"name"`
+	Stream bool   `json:"stream"`
+}
+
+type PullModelResponse struct {
+}
+
+func (c *OllamaClient) Pull(modelName string) error {
+	request := &PullModelRequest{
+		Name:   modelName,
+		Stream: false,
+	}
+	response := &PullModelResponse{}
+	req := c.httpClient.R().SetResult(&response).SetBody(request)
+	if _, err := req.Post("/api/pull"); err != nil {
+		return err
+	}
+	return nil
+
+}
+
+type OllamaChatMessage struct {
+	Role    string   `json:"role"`
+	Content string   `json:"content"`
+	Images  []string `json:"images,omitempty"`
+}
+
+type OllamaOptions struct {
+	Temperature float32 `json:"temperature,omitempty"`
+	TopK        int     `json:"top_k,omitempty"`
+	Seed        int     `json:"seed,omitempty"`
+}
+
+type ChatRequest struct {
+	Model    string              `json:"model"`
+	Messages []OllamaChatMessage `json:"messages"`
+	Stream   bool                `json:"stream"`
+	Options  OllamaOptions       `json:"options"`
+}
+
+type ChatResponse struct {
+	Model              string            `json:"model"`
+	CreatedAt          string            `json:"created_at"`
+	Message            OllamaChatMessage `json:"message"`
+	Done               bool              `json:"done"`
+	DoneReason         string            `json:"done_reason"`
+	TotalDuration      int               `json:"total_duration"`
+	LoadDuration       int               `json:"load_duration"`
+	PromptEvalCount    int               `json:"prompt_eval_count"`
+	PromptEvalDuration int               `json:"prompt_eval_duration"`
+	EvalCount          int               `json:"eval_count"`
+	EvalDuration       int               `json:"eval_duration"`
+}
+
+func (c *OllamaClient) Chat(request ChatRequest) (ChatResponse, error) {
+	response := ChatResponse{}
+	if !c.CheckModelAvailability(request.Model) {
+		if c.autoPull {
+			err := c.Pull(request.Model)
+			if err != nil {
+				return response, fmt.Errorf("error when auto pulling model %v", err)
+			}
+		} else {
+			return response, fmt.Errorf("model %s is not available", request.Model)
+		}
+	}
```

Comment:
nit
```
	if !c.CheckModelAvailability(request.Model) {
		if c.autoPull {
			err := c.Pull(request.Model)
			if err != nil {
				return response, fmt.Errorf("error when auto pulling model %v", err)
			}
		} else {
			return response, fmt.Errorf("model %s is not available", request.Model)
		}
	}
```

if it is better in terms of readability, please change it.
You can decide it.

```
isAvailable := c.CheckModelAvailability(request.Model)

if !isAvailable && !c.autoPull {
  return response, fmt.Errorf("model %s is not available", request.Model)
}

if !isAvailable {
  err := c.Pull(request.Model)
  if err != nil {
    return response, fmt.Errorf("error when auto pulling model %v", err)
  }
}

Reply:
fixed, I think it is indeed better.